Google.Cloud.EntityFrameworkCore.Spanner.QuickStart/QuickStart.cs

```diff
@@ -0,0 +1,163 @@
+using Google.Api.Gax;
+using Google.Cloud.EntityFrameworkCore.Spanner.Extensions;
+using Google.Cloud.EntityFrameworkCore.Spanner.Metadata;
+using Microsoft.EntityFrameworkCore;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading.Tasks;
+
+public class Singer
+{
+    public Singer()
+    {
+        Albums = new HashSet<Album>();
+    }
+
+    public Guid SingerId { get; set; }
+    public string FirstName { get; set; }
+    public string LastName { get; set; }
+    // FullName is generated automatically from FirstName and LastName.
+    public string FullName { get; set; }
+
+    public virtual ICollection<Album> Albums { get; set; }
+}
+
+public class Album
+{
+    public Album()
+    {
+        Tracks = new HashSet<Track>();
+    }
+
+    public Guid AlbumId { get; set; }
+    public string Title { get; set; }
+    public virtual Singer Singer { get; set; }
+
+    public virtual ICollection<Track> Tracks { get; set; }
+}
+
+public class Track
+{
+    public Track()
+    {
+    }
+
+    public Guid AlbumId { get; set; }
+    public long TrackId { get; set; }
+    public string Title { get; set; }
+
+    public virtual Album Album { get; set; }
+}
+
+public class MusicDbContext : DbContext
+{
+    private readonly string _connectionString;
+
+    public MusicDbContext(string connectionString)
+    {
+        _connectionString = connectionString;
+    }
+
+    public MusicDbContext(string connectionString, DbContextOptions<MusicDbContext> options)
+        : base(options)
+    {
+        _connectionString = connectionString;
+    }
+
+    public virtual DbSet<Singer> Singers { get; set; }
+    public virtual DbSet<Album> Albums { get; set; }
+    public virtual DbSet<Track> Tracks { get; set; }
+
+    protected override void OnConfiguring(DbContextOptionsBuilder options)
+        // Configure Entity Framework to use a Cloud Spanner database.
+        => options.UseSpanner(_connectionString);
+
+    protected override void OnModelCreating(ModelBuilder modelBuilder)
+    {
+        modelBuilder.Entity<Album>(entity =>
+        {
+            entity.HasKey(entity => new { entity.AlbumId });
+        });
+
+        modelBuilder.Entity<Singer>(entity =>
+        {
+            // FullName is generated by Cloud Spanner on each add or update.
+            entity.Property(e => e.FullName).ValueGeneratedOnAddOrUpdate();
+        });
+
+        modelBuilder.Entity<Track>(entity =>
+        {
+            // Track is INTERLEAVED IN PARENT Album.
+            entity
+                .InterleaveInParent(typeof(Album), OnDelete.Cascade)
+                .HasKey(entity => new { entity.AlbumId, entity.TrackId });
+        });
+    }
+}
+
+public static class QuickStart
+{
+    static void Main(string[] args)
+    {
+        string connectionString = "Data Source=projects/my-project/instances/my-instance/databases/my-database";
```

Comment:
nit: Maybe add a small README with a link to this line, and a short comment with a TODO here that the user should edit this line to reference their own database.